tests/test_component.py

```diff
@@ -81,26 +82,64 @@ class Media:
                 css = "style.css"
                 js = "script.js"
 
-        comp = SimpleComponent("simple_component")
-        context = Context(comp.get_context_data(variable="test"))
+        rendered = SimpleComponent.render(kwargs={"variable": "test"})
+        self.assertHTMLEqual(
+            rendered,
+            """
+            Variable: <strong>test</strong>
+            """,
+        )
+
+    @parametrize_context_behavior(["django", "isolated"])
+    def test_template_string_dynamic(self):
```

Comment:
Added missing tests for `Component.get_template_string`, `template_name`,  and `get_template`.